test/unit/io/detail/misc_test.cpp

```diff
@@ -70,3 +74,52 @@ TEST(misc, valid_compression_extensions)
         EXPECT_TRUE(std::find(valid_compression.begin(), valid_compression.end(), "zst") != valid_compression.end());
     #endif
 }
+
+template <typename compression_t>
+class misc : public ::testing::Test
+{
+public:
+    static inline std::string const compression_extension = "." + compression_t::file_extensions[0];
+    static constexpr auto expected_magic_header = compression_t::magic_header;
+    static constexpr size_t biggest_magic_header_size = seqan3::detail::bgzf_compression::magic_header.size();
+
+    std::filesystem::path file_path() const
+    {
+        return tmp_file.get_path();
+    }
+
+private:
+    std::string const tmp_file_name = "io_misc_test.txt" + compression_extension;
+    seqan3::test::tmp_filename const tmp_file{tmp_file_name.c_str()};
+};
+
+using compression_types = ::testing::Types<seqan3::detail::gz_compression,
+                                           seqan3::detail::bgzf_compression,
+                                           seqan3::detail::bz2_compression>;
+
+TYPED_TEST_SUITE(misc, compression_types, );
+
+// https://github.com/seqan/seqan3/issues/2455
+TYPED_TEST(misc, issue2455)
+{
+    {
+        auto file_path = this->file_path();
+        std::ofstream filestream{file_path};
+        auto stream_ptr = seqan3::detail::make_secondary_ostream(filestream, file_path);
+        *stream_ptr << std::string(this->biggest_magic_header_size, 'a') << '\n';
+    }
+
+    std::ifstream filestream{this->file_path()};
+    std::array<char, this->biggest_magic_header_size> magic_header{};
+    std::copy_n(std::istreambuf_iterator{filestream}, this->biggest_magic_header_size, magic_header.begin());
+
+    if constexpr (std::same_as<TypeParam, seqan3::detail::bgzf_compression>)
+    {
+        EXPECT_TRUE(TypeParam::validate_header(std::span{magic_header}));
+    }
+    else
+    {
+        EXPECT_TRUE(seqan3::detail::starts_with(magic_header, this->expected_magic_header));
+        EXPECT_FALSE(seqan3::detail::bgzf_compression::validate_header(std::span{magic_header}));
+    }
+}
```

Comment:
I think it would be easier to understand if this would be 3 test cases instead of 1 typed test as they have something in common, but are also quite different.

I mean the original problem was only gzip <=> bgzf

Reply:
Yes, it's better when it's split.
I did some more convoluted stuff before, and it did save a lot of code to have a fixture.
Then I managed to simplify everything and did not re-evaluate the usage of a fixture